crypto/evp/evp_fetch.c

```diff
@@ -81,36 +81,43 @@ static OSSL_METHOD_STORE *get_evp_method_store(OSSL_LIB_CTX *libctx)
 /*
  * To identify the method in the EVP method store, we mix the name identity
  * with the operation identity, under the assumption that we don't have more
- * than 2^24 names or more than 2^8 operation types.
+ * than 2^23 names or more than 2^8 operation types.
```

Comment:
Why the loss of the bit?

Reply:
To avoid sign extension when downshifting.
Shouldn't matter that much, I would say that it would be a challenge to produce 2^23 names too :wink:

Reply:
Shifting an unsigned value right is a logical shift.

Reply:
Yeah, but it's passed to `ossl_method_store_cache_set()`, which takes it as an `int`, which in turn affects `filter_on_operation_id()` further down.

It's possible that we should change `ossl_method_store_cache_set()`, but that may explode, so I'd rather have that done in a different PR.

Reply:
My thought he was why bother packing this at all.

I didn't think it important enough to comment on -- 8,000,000+ names ought to be sufficient for a while.

Reply:
> My thought he was why bother packing this at all.

Because this needs to be passed as the "nid" (which isn't the same as the NIDs) to `ossl_method_store_cache_set()`, and this method ID is specific to EVP fetch functions.  The OSSL_DECODER, OSSL_ENCODER and OSSL_STORE_LOADER fetching is simple, there is no operation ID to account for.

Reply:
This explanation should be a comment in the code...

Reply:
Okie, comment added